rockcraft/models/project.py

```diff
@@ -102,6 +103,136 @@ def _validate_platform_set(cls, values: Mapping[str, Any]) -> Mapping[str, Any]:
         return values
 
 
+class BuildPlanner(BaseBuildPlanner):
+
+    platforms: Dict[str, Any]
+    base: Literal["bare", "ubuntu@20.04", "ubuntu@22.04"]
+    build_base: Optional[Literal["ubuntu@20.04", "ubuntu@22.04"]]
+
+    @pydantic.validator("build_base", always=True)
+    @classmethod
+    def _validate_build_base(cls, build_base: Optional[str], values: Any) -> str:
```

Comment:
would probably need to move this validation code somewhere else so that both `Project` and `BuildPlanner` can use it